API/Controllers/AccountController.cs

```
@@ -72,6 +75,38 @@ public AccountController(UserManager<AppUser> userManager,
         _emailService = emailService;
         _eventHub = eventHub;
         _localizationService = localizationService;
+        _oidcService = oidcService;
+    }
+
+    /// <summary>
+    /// Returns the current user, as it would from login
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="UnauthorizedAccessException"></exception>
+    /// <remarks>If <see cref="OidcConfigDto.SyncUserSettings"/> is enabled, and the user </remarks>
+    /// <remarks>Updates the last active date for the user</remarks>
+    [HttpGet]
+    public async Task<ActionResult<UserDto>> GetCurrentUserAsync()
+    {
+        var user = await _unitOfWork.UserRepository.GetUserByIdAsync(User.GetUserId(), AppUserIncludes.UserPreferences | AppUserIncludes.SideNavStreams);
+        if (user == null) throw new UnauthorizedAccessException();
+
+        if (user.IdentityProvider == IdentityProvider.OpenIdConnect)
+        {
+            var oidcSettings = (await _unitOfWork.SettingsRepository.GetSettingsDtoAsync()).OidcConfig;
+            await _oidcService.SyncUserSettings(oidcSettings, User, user);
+        }
+
+        var roles = await _userManager.GetRolesAsync(user);
+        if (!roles.Contains(PolicyConstants.LoginRole)) return Unauthorized(await _localizationService.Translate(user.Id, "disabled-account"));
+
+        // Update LastActive on account
+        user.UpdateLastActive();
```

Can we wrap this in a try/catch and log/swallow the exception? On some users machines, the db throws an exception on updating last login which propagates to UI.